HTWDD/Components/Canteen/Models/Canteen.swift

```diff
@@ -8,87 +8,47 @@
+                                                                               longitude: 13.7318803))
+    static let all = [reichenbachstrasse]
 
-        var imageId: Int? {
-            switch self {
-            case .reichenbachstrasse: return 9
-            }
+    static func with(id: Id) throws -> Canteen {
+        guard let canteen = all.filter({ $0.id == id }).first else {
+            throw Canteen.Error.incompatibleCanteen(id: id)
         }
+        return canteen
     }
 
-    let id: Id
-    let name: String
-    let address: String
-    let coordinate: CLLocationCoordinate2D?
-
-    private let canteen: OpenMensaKit.Canteen
+    enum Id: String {
+        case reichenbachstrasse = "mensa_reichenbachstraße"
 
-    init?(_ openMensaCanteen: OpenMensaKit.Canteen) {
-        guard let id = Id(rawValue: openMensaCanteen.id) else {
-            return nil
+        var imageId: Int? {
+            switch self {
+            case .reichenbachstrasse: return 9
+            }
         }
-        self.id = id
-        self.name = openMensaCanteen.name
-        self.address = openMensaCanteen.address
-        self.coordinate = openMensaCanteen.coordinate
-
-        self.canteen = openMensaCanteen
     }
-
 }
 
 extension Canteen {
-
     enum Error: Swift.Error {
-        case incompatibleCanteen(id: Int)
-    }
-
-    static func get(id: Id) -> Observable<Canteen> {
-        return Observable.create { observer in
-
-            OpenMensaKit.Canteen.get(withID: id.rawValue) { result in
-                switch result {
-                case .success(let canteen):
-                    guard let c = Canteen(canteen) else {
-                        observer.onError(Error.incompatibleCanteen(id: canteen.id))
-                        return
-                    }
-                    observer.onNext(c)
-                    observer.onCompleted()
-                case .failure(let err):
-                    observer.onError(err)
-                }
-            }
-
-            return Disposables.create()
-        }
+        case incompatibleCanteen(id: Canteen.Id)
     }
 
-    func getMeals(date: Date) -> Observable<[Meal]> {
-        return Observable.create { observer in
-
-            self.canteen.getMeals(forDay: date, completion: { result in
-                switch result {
-                case .success(let meals):
-                    let mapped = meals.map { m in
-                        return Meal(canteen: self.id, date: date, openMensaMeal: m)
-                    }
-                    observer.onNext(mapped)
-                    observer.onCompleted()
-                case .failure(let err):
-                    observer.onError(err)
-                }
-            })
-
-            return Disposables.create()
-        }
+    func getMeals() -> Observable<[Meal]> {
+        // FIXME: Replace with self-hosted URL
+        let escapedID = self.id.rawValue.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)!
```

Comment:
Please do not force unwrap, then better use the rawValue as fallback

Reply:
`nil` here shouldn't be possible on "normal" string values, see [here](https://stackoverflow.com/questions/33558933/why-is-the-return-value-of-string-addingpercentencoding-optional#33558934). I did update it to use the raw value as a fallback though.